src/ast.ml

```diff
@@ -39,28 +54,45 @@ type link_kind =
   | Img
   | Url
 
-type inline =
+type emph =
+  {
+    kind: emph_kind;
+    style: emph_style;
+    md: inline;
+  }
```

Comment:
One possibility is to use different modules to group types for each node type, e.g.:
```
module Emph = struct
  type inline
  type kind = (* definition of type emph_kind *)
  type style = (* emph_style *)
  type 'inline t =
    { style: style;
      kind: kind;
      content: 'inline }
end 
```
Similarly for modules `Link`, `Ref`, `Code_block` and `List`. The type argument to `t` is not very nice but it is necessary to break the recursion.

What do you think?

Reply:
Seems good, I will give it a try (sorry for the delay)

Reply:
Sure, no worries! Am not 100% this is the best design, but I feel it is worth a try.

Reply:
I had no time to do it last week, but it's done now. It looks like it's working properly, and it cleans a few things.

I removed the `type inline`, I did not see the interest of it